### PR TITLE
[libbeat] Fix encoding and file offset issues in the disk queue

### PR DESCRIPTION
**Pull request by faec**
## What does this PR do?

This PR fixes https://github.com/elastic/beats/issues/26335 (incorrect decoding of multi-byte characters) by changing the disk queue serialization format to CBOR rather than JSON. It also fixes some file position calculations that weren't properly updated along with the schema change in https://github.com/elastic/beats/pull/22970.

There are no user-visible changes in this PR other than the queue working correctly on new segment files; the underlying schema is altered, but files created by 7.13 or earlier will deserialize with JSON as before.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Review by fearful-symmetry — APPROVED** [review]
I only _partly_ understand the surrounding code, but the PR itself LGTM.

File: libbeat/publisher/queue/diskqueue/serialize_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/publisher"
+)
+
+// A test to make sure serialization works correctly on multi-byte characters.
+func TestSerializeMultiByte(t *testing.T) {
+	asciiOnly := "{\"name\": \"Momotaro\"}"
+	multiBytes := "{\"name\": \"桃太郎\"}"
```
Comment (urso) [inline libbeat/publisher/queue/diskqueue/serialize_test.go:33]: let's create a table driven test and separate the 2 cases into separate tests.

File: libbeat/publisher/queue/diskqueue/writer_loop.go [hunk 2]
```diff
@@ -173,6 +173,9 @@ outerLoop:
 				// This can only happen if the queue is being closed; abort.
 				break
 			}
+			// We're creating a new segment file, set the initial bytes written
+			// to the header size.
+			curSegmentResponse.bytesWritten = wl.currentSegment.headerSize()
```
Comment (urso) [inline libbeat/publisher/queue/diskqueue/writer_loop.go:178]: This looks like a bug fix. Did we introduce the problem when moving to absolute adresses, or does it fix an issue we have had before? In the later case we might want to add a changelog entry (and maybe a separate PR).

Comment (faec) [reply libbeat/publisher/queue/diskqueue/writer_loop.go:178]: No, this was just an oversight in the previous (absolute address) PR, so it hasn't hit a release.

File: libbeat/publisher/queue/diskqueue/serialize_test.go (hunk 1 above)
Comment (faec) [reply libbeat/publisher/queue/diskqueue/serialize_test.go]: done

**Review by urso — APPROVED** [review]

**Pull request merged**